### DISCUSSION TITLE
How to test an async setups with props?

### DISCUSSION THREAD

**Discussion by BIGScream** [vuejs/test-utils#1876] [Q&A] [answered]
In the Documentation, the Code for an async setup is the following:

```
test('Async component', () => {
  const TestComponent = defineComponent({
    components: { Async },
    template: '<Suspense><Async/></Suspense>'
  })

  const wrapper = mount(TestComponent)
  // ...
})
```

That works perfectly. But what if, i want to set props of "Async"? 

I cant' set them via wrapper.vm.setProps and i cant set them in the mount(...) Function, because these are the props from TestComponent. I also tried to write it into the Template, as in usual Vue Templates - that results, that the Object is converted to a String...

**Comment by cexbrayat** [comment, marked as answer]
If I understand your question correctly, I would do something like that:

```ts
test('Async component', () => {
  const TestComponent = defineComponent({
    components: { Async },
    template: '<Async :something="something" />',
    props: { something: Object }
  })

  const wrapper = mount(TestComponent, { props: { something: { hello: 'world' } })
})
```

**Reply by BIGScream** [reply]
thank you :) works fine, only one curly brace too few in mount :)

**Reply by cexbrayat** [reply]
Haha great! yeah that was untested pseudo-code 😉
